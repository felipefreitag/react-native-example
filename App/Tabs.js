import React, { Component } from 'react'
import { View } from 'react-native'

import Navigation from './Navigation'
import Future from './Future'
import Joined from './Joined'
import Past from './Past'

export default class Tabs extends Component {
  constructor(props) {
    super(props)
    this.state = {
      active: 'future',
    }
    this.handleNavigate = this.handleNavigate.bind(this)
  }

  handleNavigate(tab) {
    this.setState({
      active: tab,
    })
  }

  render() {
    const { active } = this.state

    return (
      <View style={{ flex: 3 }}>
        <Navigation active={active} handleNavigate={this.handleNavigate} />
        {active === 'future' && <Future />}
        {active === 'joined' && <Joined />}
        {active === 'past' && <Past />}
      </View>
    )
  }
}
