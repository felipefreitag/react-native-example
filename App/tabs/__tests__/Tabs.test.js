import React from 'react'
import renderer from 'react-test-renderer'
import Tabs from '../Tabs'

const user = { futureTrips: [] }

it('renders correctly', () => {
  const tree = renderer.create(<Tabs user={user} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('handles state correctly', () => {
  const instance = renderer.create(<Tabs user={user} />).getInstance()
  expect(instance.state.active).toBe('future')
  instance.handleNavigate('foo')
  expect(instance.state.active).toBe('foo')
})
