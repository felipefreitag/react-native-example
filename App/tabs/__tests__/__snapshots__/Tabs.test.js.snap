// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 6,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "space-evenly",
        "paddingHorizontal": 50,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderBottomColor": "#5B97C8",
          "borderBottomWidth": 3,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        FUTURE
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderBottomColor": "lightgray",
          "borderBottomWidth": 3,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        JOINED
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderBottomColor": "lightgray",
          "borderBottomWidth": 3,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        PAST
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView>
      <View />
    </RCTScrollView>
  </View>
</View>
`;
